src/Bundle/Maker/MakeFactory.php

```diff
@@ -234,14 +264,43 @@ private function defaultPropertiesFor(string $class, bool $allFields): iterable
             $value = "null, // TODO add {$type} {$dbType} type manually";
             $length = $property['length'] ?? '';
 
-            if (\array_key_exists($type, self::DEFAULTS)) {
-                $value = self::DEFAULTS[$type];
+            if (\array_key_exists($type, self::DEFAULTS_FOR_PERSISTED)) {
+                $value = self::DEFAULTS_FOR_PERSISTED[$type];
             }
 
             yield $property['fieldName'] => \str_replace('{length}', (string) $length, $value);
         }
     }
 
+    /**
+     * @param class-string $class
+     */
+    private function defaultPropertiesForNotPersistedObject(string $class, bool $allFields): iterable
+    {
+        $object = new \ReflectionClass($class);
+
+        foreach ($object->getProperties() as $property) {
+            // ignore identifiers and nullable fields
+            if (!$allFields && ($property->hasDefaultValue() || !$property->hasType() || $property->getType()?->allowsNull())) {
```

Comment:
this could be a first draft as I'm not sure how far we should go here:
- should we look at the constructor to know which properties are mandatory?
- should we activate `alwaysForceProperties` if a property has type, no default value and is not present in construjctor?
- should we look at phpdoc?

Reply:
Let's leave it simple to start. We could look into using `symfony/property-info`.

Reply:
not sure about `symfony/property-info`, it does not help to guess if the property has a default value

Reply:
Maybe I'd merge this PR like this if you're OK and I'll make something a little bit more evolved in another PR

Reply:
Yep!